File: app/src/main/java/com/example/turnero/Utilities.java
package com.example.turnero;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.view.MotionEvent;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.EditText;

public class Utilities {

    public static void setupUI(View view, Activity activity) {
        // si estamos no estamos en una instancia de EditText y tocamos la pantalla, el teclado se escondera.
        if(! (view instanceof EditText)) {
            // colocamos una escucha para detectar los toques en la pantalla
            view.setOnTouchListener(new View.OnTouchListener() {
                @SuppressLint("ClickableViewAccessibility")
                @Override
                // si hay toques, llamamos al metodo hideSoftKeyboard
                public boolean onTouch(View view, MotionEvent motionEvent) {
                    hideSoftKeyboard(activity);
                    return false;
                }
            });
        }
        // lo mismo si estamos en una instancia de viewgroup
        if (view instanceof ViewGroup) {
            for (int i = 0; i < ((ViewGroup) view).getChildCount(); i++) {
                View innerView = ((ViewGroup) view).getChildAt(i);
                //setupUI(innerView);
            }
        }
    }

    // metodo para esconder el teclado, usando el InputMethodManager que es la funcionalidad nativa de accesibilidad
    public static void hideSoftKeyboard(Activity activity) {
        InputMethodManager inputMethodManager = (InputMethodManager)  activity.getSystemService(Activity.INPUT_METHOD_SERVICE);
        inputMethodManager.hideSoftInputFromWindow(activity.getCurrentFocus().getWindowToken(), 0);
    }




}

File: app/src/main/java/com/example/turnero/utilities/Utilities.kt
package com.example.turnero.utilities

import android.app.Activity
import android.widget.EditText
import android.annotation.SuppressLint
import android.view.View
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import androidx.fragment.app.FragmentActivity

object Utilities {
    @SuppressLint("ClickableViewAccessibility")
    @JvmStatic
    fun setupUI(view: View, activity!!: FragmentActivity?) {
        // si estamos no estamos en una instancia de EditText y tocamos la pantalla, el teclado se escondera.
        if (view !is EditText) {
            // colocamos una escucha para detectar los toques en la pantalla
            view.setOnTouchListener(View.OnTouchListener
            // si hay toques, llamamos al metodo hideSoftKeyboard
            { _, _ ->
                hideSoftKeyboard(activity!!)
                false
            })
        }
        // lo mismo si estamos en una instancia de viewgroup
        if (view is ViewGroup) {
            for (i in 0 until view.childCount) {
                val innerView = view.getChildAt(i)
                //setupUI(innerView);
            }
        }
    }

    // metodo para esconder el teclado, usando el InputMethodManager que es la funcionalidad nativa de accesibilidad
    fun hideSoftKeyboard(activity: Activity) {
        val inputMethodManager =
            activity.getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
        inputMethodManager.hideSoftInputFromWindow(activity.currentFocus!!.windowToken, 0)
    }
}